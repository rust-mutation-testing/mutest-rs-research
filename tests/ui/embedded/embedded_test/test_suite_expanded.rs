//@ print-tests
//@ print-targets
//@ print-mutations
//@ run: fail
//@ stdout
//@ mutest-flags: --Zembedded

// HACK: Stub out the runtime dependencies of the embedded_test macro expansion.
mod embedded_test {
    pub mod export {
        pub fn check_outcome((): ()) -> ! {
            std::process::exit(0);
        }
    }
}

struct Peripherals;

fn mutable_fn(a: u32, b: u32) -> u32 {
    a + b
}

mod tests {
    use super::*;

    fn init() -> Peripherals {
        Peripherals
    }

    fn simple_test() {
        assert_eq!(5, mutable_fn(2, 3));
    }
    #[doc(hidden)]
    #[allow(unused)]
    fn __simple_test_entrypoint() -> ! {
        {
            let outcome;
            { let state = init(); outcome = simple_test(); }
            embedded_test::export::check_outcome(outcome);
        }
    }
    #[link_section = ".embedded_test.tests"]
    #[export_name = "{\"disambiguator\":10289733635752023323,\"name\":\"simple_test\",\"ignored\":false,\"should_panic\":false}"]
    static __SIMPLE_TEST_SYM: (fn() -> !, &'static str) = (__simple_test_entrypoint, "test_suite::tests");

    fn test_taking_peripherals(_peripherals: Peripherals) {
        assert_eq!(5, mutable_fn(2, 3));
    }
    #[doc(hidden)]
    #[allow(unused)]
    fn __test_taking_peripherals_entrypoint() -> ! {
        {
            let outcome;
            { let state = init(); outcome = test_taking_peripherals(state); }
            embedded_test::export::check_outcome(outcome);
        }
    }
    #[link_section = ".embedded_test.tests"]
    #[export_name = "{\"disambiguator\":10289733635752023323,\"name\":\"test_taking_peripherals\",\"ignored\":false,\"should_panic\":false}"]
    static __TEST_TAKING_PERIPHERALS_SYM: (fn() -> !, &'static str) = (__test_taking_peripherals_entrypoint, "test_suite::tests");

    fn panicking_test() {
        panic!();
    }
    #[doc(hidden)]
    #[allow(unused)]
    fn __panicking_test_entrypoint() -> ! {
        {
            let outcome;
            { let state = init(); outcome = panicking_test(); }
            embedded_test::export::check_outcome(outcome);
        }
    }
    #[link_section = ".embedded_test.tests"]
    #[export_name = "{\"disambiguator\":10289733635752023323,\"name\":\"panicking_test\",\"ignored\":false,\"should_panic\":true}"]
    static __PANICKING_TEST_SYM: (fn() -> !, &'static str) = (__panicking_test_entrypoint, "test_suite::tests");

    fn panicking_test_with_message() {
        panic!("message");
    }
    #[doc(hidden)]
    #[allow(unused)]
    fn __panicking_test_with_message_entrypoint() -> ! {
        {
            let outcome;
            { let state = init(); outcome = panicking_test_with_message(); }
            embedded_test::export::check_outcome(outcome);
        }
    }
    #[link_section = ".embedded_test.tests"]
    #[export_name = "{\"disambiguator\":10289733635752023323,\"name\":\"panicking_test_with_message\",\"ignored\":false,\"should_panic\":true}"]
    static __PANICKING_TEST_WITH_MESSAGE_SYM: (fn() -> !, &'static str) = (__panicking_test_with_message_entrypoint, "test_suite::tests");

    fn ignored_test() {}
    #[doc(hidden)]
    #[allow(unused)]
    fn __ignored_test_entrypoint() -> ! {
        {
            let outcome;
            { let state = init(); outcome = ignored_test(); }
            embedded_test::export::check_outcome(outcome);
        }
    }
    #[link_section = ".embedded_test.tests"]
    #[export_name = "{\"disambiguator\":10289733635752023323,\"name\":\"ignored_test\",\"ignored\":true,\"should_panic\":false}"]
    static __IGNORED_TEST_SYM: (fn() -> !, &'static str) = (__ignored_test_entrypoint, "test_suite::tests");
}
