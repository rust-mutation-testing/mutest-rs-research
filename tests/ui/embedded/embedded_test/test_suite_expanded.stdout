
@@@ tests @@@

test tests::ignored_test [ignored]
test tests::panicking_test
test tests::panicking_test_with_message
test tests::simple_test
test tests::test_taking_peripherals

tests: 5 total; 1 ignored

@@@ targets @@@

tests -(0)-> mutable_fn at tests/ui/embedded/embedded_test/test_suite_expanded.rs:19:1: 19:37 (#0)
  (0) tests::simple_test
  (0) tests::test_taking_peripherals

targets: 1 total; 1 safe; 0 unsafe (0 tainted)

@@@ mutations @@@

[arg_default_shadow] ignore `a` argument by shadowing it with `Default::default()` in mutable_fn at tests/ui/embedded/embedded_test/test_suite_expanded.rs:19:15: 19:21
  <-(0)- tests::simple_test
  <-(0)- tests::test_taking_peripherals

[arg_default_shadow] ignore `b` argument by shadowing it with `Default::default()` in mutable_fn at tests/ui/embedded/embedded_test/test_suite_expanded.rs:19:23: 19:29
  <-(0)- tests::simple_test
  <-(0)- tests::test_taking_peripherals

[math_op_add_mul_swap] swap operator `+` for `*` in mutable_fn at tests/ui/embedded/embedded_test/test_suite_expanded.rs:20:5: 20:10
  <-(0)- tests::simple_test
  <-(0)- tests::test_taking_peripherals

[math_op_add_sub_swap] swap operator `+` for `-` in mutable_fn at tests/ui/embedded/embedded_test/test_suite_expanded.rs:20:5: 20:10
  <-(0)- tests::simple_test
  <-(0)- tests::test_taking_peripherals

4 mutations; 4 safe; 0 unsafe (0 tainted)
